docs/proposals/versioning-and-release.md

```diff
@@ -0,0 +1,80 @@
+# Devfile Versioning and Release Process
+This design document outlines the proposed versioning and release process for the Devfile spec. 
+
+This document summarizes parts from the Devfile API technical meeting slides presented by @davidfestal back in October, and I recommend having a read through of it: https://docs.google.com/presentation/d/1ohM1HzPB59a3ajvB7rVWJkKONLBAuT0mb5P20_A6vLs/edit#slide=id.g8fc722bef9_1_8
+
+## Versioning
+
+The following sections outline how we version the Devfile Kubernetes API, as well as the JSON schema generated from the API.
+
+### Kubernetes API
+The Devfile Kubernetes API (defined in https://github.com/devfile/api/) is the single source of truth for the Devfile spec. It’s defined under `pkg/apis/` and contains the Go structs that are used by the devworkspace operator and devfile library. It's also where the Devfile JSON schema is generated from. 
+
+**Versioning Scheme**: [Kubernetes](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning) (e.g. v1alpha1, v1beta2, v1, v2, etc)
+
+**How to Update**: 
+
+   1) Add a new folder for the version in the [devfile/api](https://github.com/devfile/api/) repository under [pkg/apis/workspaces](https://github.com/devfile/api/tree/master/pkg/apis/workspaces). For example `pkg/apis/workspaces/v1` if bumping the K8S API version to `v1`.
+   2) Add a schema and version in the CRD manifests
+   3) Generate the JSON schema from the API. New JSON schema will be located under `schemas/latest`.
+   4) Update the devworkspace operator and devfile library to consume the Go structs in the new K8S API version, as needed.
+
+**When to Update?**  As incrementing the Kubernetes API version for Devfile is a relatively heavy process, and affects the library, only update the K8s API version when absolutely needed (for **big** changes or backwards incompatible changes).
+   - Backwards incompatible changes are defined as any change in the schema that would cause K8S API validation errors on the old resource version (e.g. removed fields or new mandatory fields)
```

Comment:
```suggestion
   - Backwards incompatible changes are defined as any change in the schema that would cause K8S API validation errors on the old resource version (e.g. removed fields or new mandatory fields without a default)
```